sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -1112,6 +1171,34 @@ def _plot_two_way_partial_dependence(
                 ax.set_xlabel(self.feature_names[feature_idx[0]])
             ax.set_ylabel(self.feature_names[feature_idx[1]])
 
+    def add_individual_line_color_info(self, color_dict):
+        last_ax_idx_not_none = np.argwhere(self.axes_.flatten() != None)[-1][
+            0
+        ]  # is not None, doesn't work here
+        if color_dict.get("type", None) is None:
+            return None
+        if color_dict["type"] == "categorical":
+            custom_handles = []
+            custom_labels = []
+
+            for category, color in color_dict["categorical_mapping"].items():
+                custom_handles.append(Line2D([0], [0], color=color, lw=1))
+                custom_labels.append(category)
+            # make legend disappear for all plots but the last
+            for ax in self.axes_.flatten()[:last_ax_idx_not_none]:
+                ax.legend().remove()
+            h, l = self.axes_.flatten()[
+                last_ax_idx_not_none
+            ].get_legend_handles_labels()
```

Comment:
du könntest den teil zuerst machen und dann bei deinem for loop oben die sachen direkt bei `h` und `l` appenden